smoketest/compose/auth_proxy.yml

```diff
@@ -0,0 +1,49 @@
+version: "3"
+services:
+  cryostat:
+    environment:
+      QUARKUS_HTTP_PROXY_PROXY_ADDRESS_FORWARDING: 'true'
+      QUARKUS_HTTP_PROXY_ALLOW_X_FORWARDED: 'true'
+      QUARKUS_HTTP_PROXY_ENABLE_FORWARDED_HOST: 'true'
+      QUARKUS_HTTP_PROXY_ENABLE_FORWARDED_PREFIX: 'true'
+      QUARKUS_HTTP_PROXY_TRUSTED_PROXIES: localhost:8080,auth:8080
+  auth:
+    # the proxy does not actually depend on cryostat being up, but we use this
+    # to ensure that when the smoketest tries to open the auth login page in a
+    # browser tab, it does so only after the upstream cryostat is actually
+    # available to be proxied to
+    depends_on:
+      cryostat:
+        condition: service_healthy
+    deploy:
+      resources:
+        limits:
+          cpus: '0.1'
+          memory: 32m
+    image: quay.io/oauth2-proxy/oauth2-proxy:latest
+    user: ${UID:-0}
```

Comment:
Should this default to the container's default user instead of root? https://github.com/oauth2-proxy/oauth2-proxy/blob/52ad31752d0d0b38673d419e56ddfc4cb2f56857/Dockerfile#L53C6-L53C11

Reply:
Ah yes, thanks. I was messing around with things like that before trying to deal with the sporadic permissions issues I was seeing when mounting the config files as volumes, and forgot to drop this part of the experiment.

Reply:
Though, I think when you're running with rootless docker/podman, this ends up running as the original user's UID. Within the container this appears to be root, but isn't actually.